robottelo/host_helpers/repository_mixins.py

```diff
@@ -0,0 +1,743 @@
+        ]
+
+
+class SatelliteCapsuleRepository(GenericRHRepository):
+    """Satellite capsule repository"""
+
+    _key = constants.PRODUCT_KEY_SAT_CAPSULE
+
+    @property
+    def url(self):
+        if int(self.distro) == constants.SATELLITE_OS_VERSION:
+            return settings.repos.capsule_repo
+        return None
+
+
+class VirtualizationAgentsRepository(GenericRHRepository):
+    """Virtualization Agents repository"""
+
+    _key = constants.PRODUCT_KEY_VIRT_AGENTS
+    _distro = constants.DISTRO_RHEL6
+
+
+class RHELCloudFormsTools(GenericRHRepository):
+    _distro = constants.DISTRO_RHEL6
+    _key = constants.PRODUCT_KEY_CLOUD_FORMS_TOOLS
+
+
+class RHELAnsibleEngineRepository(GenericRHRepository):
+    """Red Hat Ansible Engine Repository"""
+
+    _key = constants.PRODUCT_KEY_ANSIBLE_ENGINE
+
+
+class RepositoryCollection:
+    """Repository collection"""
+
+    _distro = None
+    _org = None
+    _items = []
+    _repos_info = []
+    _custom_product_info = None
+    _os_repo = None
+    _setup_content_data = None
+    satellite = None
+
+    def __init__(self, distro=None, repositories=None):
+
+        self._items = []
+
+        if distro is not None and distro not in constants.DISTROS_SUPPORTED:
+            raise DistroNotSupportedError(f'distro "{distro}" not supported')
+        if distro is not None:
+            self._distro = distro
+
+        if repositories is None:
+            repositories = []
+        self.add_items(repositories)
+
+    @property
+    def distro(self):
+        return self._distro
+
+    @property
+    def repos_info(self):
+        return self._repos_info
+
+    @property
+    def custom_product(self):
+        return self._custom_product_info
+
+    @property
+    def os_repo(self):
+        return self._os_repo
+
+    @os_repo.setter
+    def os_repo(self, repo):
+        if self.os_repo is not None:
+            raise OnlyOneOSRepositoryAllowed('OS repo already added.(Only one OS repo allowed)')
+        if not isinstance(repo, RHELRepository):
+            raise ValueError(f'repo: "{repo}" is not an RHEL repo')
+        self._os_repo = repo
+
+    @property
+    def repos_data(self):
+        return [repo.data for repo in self]
+
+    @property
+    def rh_repos(self):
+        return [item for item in self if item.cdn]
+
+    @property
+    def custom_repos(self):
+        return [item for item in self if not item.cdn]
+
+    @property
+    def rh_repos_info(self):
+        return [
+            repo_info for repo_info in self._repos_info if repo_info['red-hat-repository'] == 'yes'
+        ]
+
+    @property
+    def custom_repos_info(self):
+        return [
+            repo_info for repo_info in self._repos_info if repo_info['red-hat-repository'] == 'no'
+        ]
+
+    @property
+    def setup_content_data(self):
+        return self._setup_content_data
+
+    @property
+    def need_subscription(self):
+        if self.rh_repos:
```

Comment:
`return bool(self.rh_repos)`   - but that begs the question, why does this property exist when `self.rh_repos` is the same thing?